ecosystem/typescript/sdk/src/providers/aptos_client.ts

```diff
@@ -233,6 +233,25 @@ export class AptosClient {
     return data;
   }
 
+  /**
+   * Queries a 0x1::object::Object resource associated with given address
+   * @param address Hex-encoded 32 byte Aptos account address
+   * @param query.ledgerVersion Specifies ledger version of transactions. By default latest version will be used
+   * @returns 0x1::object::Object resource
+   */
+    @parseApiError
+    async getObjectResource(
+      address: MaybeHexString,
+      query?: { ledgerVersion?: AnyNumber },
+    ): Promise<Gen.MoveResource> {
+      const resources = await this.getAccountResources(address, query);
+      if(resources.length !== 2 || resources.find((r) => r.type === "0x1::object::ObjectCore") === undefined) {
```

Comment:
Let's clean this up.

The resources can be > 2 and also I'm not sure there's a guarantee that 0x1::object::ObjectCore will be in the first page of getAccountResources.

Let's instead use `getAccountResource` and get specifically the `0x1::object::ObjectCore` resource.

Reply:
@gregnazario Sorry, I think what we want from this `getObjectResource` interface is not the `0x1::object::ObjectCore` resource. What we need is the `0x1::object::Object<T0>` resource. With the `address` as input arguments, we do not know the `T0`. So we cannot use `getAccountResource` to get the `Object` resource. Any idea?

You can see this example:
https://fullnode.testnet.aptoslabs.com/v1/accounts/0xb584c623d565f87cc3c20df8fad99b9a1d2783d00e1ff00236faa05190a7ae4f/resources

Reply:
I see, there's a misunderstanding of how Objects work.

An object would be all of the applicable resources, and could be extended.

You could have an object as a resource group with 3 different resources say:
1. Cat
2. Dog
3. Monkey

You could access that address (the same object) as one of three types:
1. Object<Cat>
2. Object<Dog>
3. Object<Monkey>

This allows the flexibility of bundling these together

Reply:
Changed.